controllers/topology/internal/mergepatch/mergepatch_test.go

```diff
@@ -100,6 +100,34 @@ func TestNewHelper(t *testing.T) {
 			wantHasSpecChanges: false,
 			wantPatch:          []byte("{\"metadata\":{\"labels\":{\"foo\":\"bar\"}}}"),
 		},
+		{
+			name: "Field (metadata.label) align to modified when path is authoritative",
+			original: &unstructured.Unstructured{ // current
+				Object: map[string]interface{}{
+					"metadata": map[string]interface{}{
+						"labels": map[string]interface{}{
+							"a": "a",
+							"b": "b-changed",
+						},
+					},
+				},
+			},
+			modified: &unstructured.Unstructured{ // desired
+				Object: map[string]interface{}{
+					"metadata": map[string]interface{}{
+						"labels": map[string]interface{}{
+							// a missing
+							"b": "b",
+							"c": "c",
+						},
+					},
+				},
+			},
+			options:            []HelperOption{AuthoritativePaths{contract.Path{"metadata", "labels"}}},
```

Comment:
Maybe add the inverse of this test to ensure the direction of the patch.

Reply:
it is already covered by other tests but added one specific for metadata